src/main/java/teammates/ui/webapi/action/CreateFeedbackResponseAction.java

```diff
@@ -96,12 +96,17 @@ public ActionResult execute() {
         }
 
         FeedbackResponseCreateRequest createRequest = getAndValidateRequestBody(FeedbackResponseCreateRequest.class);
+        String recipientSection = getRecipientSection(feedbackQuestion.getCourseId(),
+                feedbackQuestion.getRecipientType(), createRequest.getRecipientIdentifier());
+        if (intent == Intent.INSTRUCTOR_SUBMISSION && feedbackQuestion.giverType == FeedbackParticipantType.INSTRUCTORS
```

Comment:
The logic should be in `getRecipientSection`? and we don't need to check `intent`?

Reply:
`SELF` being mapped to `Const.DEFAULT_SECTION` is not right for student giver type.

Reply:
I mean we should put the logic inside `getRecipientSection` :P

Reply:
Intent is necessary to sieve out student-instructors.

Reply:
Actually if we don't remove the check for `intent`, the case you mention will cause exactly the problem you have fixed.

For student-instructor path, the recipient section should be 'None' rather than student's section.

Reply:
I meant it's to sieve out students and instructors for entities that are both. Student-instructor feedback path is common and doesn't belong here.